ddtrace/opentracer/tracer.py

```diff
@@ -71,35 +75,123 @@ def __init__(self, service_name=None, config=None, scope_manager=None):
+            tracer.start_span(
+                '...',
+                child_of=parent_span)
+        Starting a child Span in a more verbose way::
+            tracer.start_span(
+                '...',
+                references=[opentracing.child_of(parent_span)])
+        :param operation_name: name of the operation represented by the new
+            span from the perspective of the current service.
+        :param child_of: (optional) a Span or SpanContext instance representing
+            the parent in a REFERENCE_CHILD_OF Reference. If specified, the
+            `references` parameter must be omitted.
+        :param references: (optional) a list of Reference objects that identify
+            one or more parent SpanContexts. (See the Reference documentation
+            for detail)
+        :param tags: an optional dictionary of Span Tags. The caller gives up
+            ownership of that dictionary, because the Tracer may use it as-is
+            to avoid extra data copying.
+        :param start_time: an explicit Span start time as a unix timestamp per
+            time.time()
+        :param ignore_active_span: an explicit flag that ignores the current
+            active `Scope` and creates a root `Span`.
+        :return: an already-started Span instance.
+        """
+        ot_parent = child_of      # 'ot_parent' is more readable than 'child_of'
+        ot_parent_context = None  # the parent span's context
+        dd_parent = None          # the child_of to pass to the ddtracer
+
+        # Okay so here's the deal for ddtracer.start_span:
+        #  - whenever child_of is not None ddspans with parent-child relationships
+        #    will share a ddcontext which maintains a hierarchy of ddspans for
+        #    the execution flow
+        #  - when child_of is a ddspan then the ddtracer uses this ddspan to create
+        #    the child ddspan
+        #  - when child_of is a ddcontext then the ddtracer uses the ddcontext to
+        #    get_current_span() for the parent
+        if ot_parent is None and not ignore_active_span:
+            # attempt to get the parent span from the scope manager
+            scope = self._scope_manager.active
+            parent_span = getattr(scope, 'span', None)
+            ot_parent_context = getattr(parent_span, 'context', None)
+            # we want the ddcontext of the active span in order to maintain the
+            # ddspan hierarchy
+            dd_parent = getattr(ot_parent_context, '_dd_context', None)
+        elif ot_parent is not None and isinstance(ot_parent, Span):
+            # a span is given to use as a parent
+            ot_parent_context = ot_parent.context
+            dd_parent = ot_parent._dd_span
+        elif ot_parent is not None and isinstance(ot_parent, SpanContext):
+            # a span context is given to use to find the parent ddspan
+            dd_parent = ot_parent._dd_context
+        elif ot_parent is None:
```

Comment:
you can remove this block since it's a `pass` and in the docstring (or in a generic comment) must be clear what we're doing

Reply:
The pass is so that we don't fall through to the `else` condition.